fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/DimensionFieldMapper.java

```diff
@@ -0,0 +1,32 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.dimension;
+
+import java.util.Locale;
+
+/**
+ * Maps a Dimension Field value to a string.
+ */
+@FunctionalInterface
+public interface DimensionFieldMapper {
+
+    /**
+     * A converter function for the dimension field Mapping.
+     *
+     * @param dimension The dimension object used to configure the Dimension
+     * @param dimensionField The dimension field object
+     *
+     * @return the converted string for the dimension field
+     */
+    String converter(Dimension dimension, DimensionField dimensionField);
+
```

Comment:
This is either 'map' if you generically want to indicate that this is a mapping operation, or, given the fact that this is a DimensionFieldMapper a more meaningful name like dimensionFieldName or something scoped to the meaning.